### ISSUE TITLE
no need to copy data when func (mp *inmemoryPart) Init

### ISSUE THREAD

**Issue by ahfuzhang** [VictoriaMetrics/VictoriaMetrics#2247]
see: https://github.com/VictoriaMetrics/VictoriaMetrics/blob/99ee1c13c05479d974d496c8b44c5c6b84ed03b0/lib/mergeset/inmemory_part.go#L61
and line 65

there is no need to copy compressed data in storageBlock.

we can add if when use itemsData and lensData:
```go
buf := mp.itemsData.B
if len(buf)==0{
    buf := mp.sb.itemsData
}
```

**Comment by valyala** [comment]
@ahfuzhang , thanks for spotting this inefficient code! It has been addressed in commits e8fdb27625a2985a23f11332683d3a99f1e53832 , c84a8b34cc64376fc37ede87c8f22d6450956022  and 0a4aadffacacc4c512931e2385fe320de6432149

**Comment by valyala** [comment]
All the commits mentioned above have been included in [v1.74.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). Closing the issue as fixed.
